scripts/upgrade

```diff
@@ -6,113 +6,148 @@
+        ynh_app_setting_set --app=$app --key=final_path --value=$final_path
     fi
 fi
 
+#=================================================
+# BACKUP BEFORE UPGRADE THEN ACTIVE TRAP
+#=================================================
+ynh_script_progression --message="Backing up the app before upgrading (may take a while)..." --weight=6
+
+# Backup the current version of the app
+ynh_backup_before_upgrade
+ynh_clean_setup () {
+	# restore it if the upgrade fails
+	ynh_restore_upgradebackup
+}
+# Exit if an error occurs during the execution of the script
+ynh_abort_if_errors
+
 #=================================================
 # CHECK THE PATH
 #=================================================
 
 # Normalize the URL path syntax
-path_url=$(ynh_normalize_url_path $path_url)
+path_url=$(ynh_normalize_url_path --path_url=$path_url)
 
 #=================================================
 # STANDARD UPGRADE STEPS
 #=================================================
 # DOWNLOAD, CHECK AND UNPACK SOURCE
 #=================================================
 
-# Download, check integrity, uncompress and patch the source from app.src
-# For this app sources are in app subdirectory 
-tmp_dir=$(mktemp -d)
-ynh_setup_source "$tmp_dir"
-# Dist is the compiled version. Other directories are source code and tools to build the program
-cp -R "$tmp_dir/dist/."  "$final_path/"
-# Imgur is deactivated on the specific upstream selected, but this text is still here
-# Don't use a patch because it's minified js
-ynh_replace_string "are currently uploaded to imgur.*re working on changing this" "size is limited" "$final_path/scripts/amd-app.js"
-ynh_secure_remove "$tmp_dir"
+if [ "$upgrade_type" == "UPGRADE_APP" ]
+then
+	ynh_script_progression --message="Upgrading source files..." --weight=4
+
+	# Download, check integrity, uncompress and patch the source from app.src
+	# For this app sources are in app subdirectory 
+	tmp_dir=$(mktemp -d)
+
+	ynh_setup_source --dest_dir="$tmp_dir"
+	# Dist is the compiled version. Other directories are source code and tools to build the program
+	cp -a "$tmp_dir/dist/."  "$final_path/"
+	ynh_secure_remove --file="$tmp_dir"
+
+	# Imgur is deactivated on the specific upstream selected, but this text is still here
+	# Don't use a patch because it's minified js
+	ynh_replace_string --match_string="are currently uploaded to imgur.*re working on changing this" --replace_string="size is limited" --target_file="$final_path/scripts/amd-app.js"
+fi
 
 #=================================================
 # NGINX CONFIGURATION
 #=================================================
+ynh_script_progression --message="Upgrading nginx web server configuration..."
 
 # Create a dedicated nginx config
 ynh_add_nginx_config
 
+#=================================================
+# SPECIFIC UPGRADE
+#=================================================
+# PATCH SOURCE
+#=================================================
+
+if [ "$upgrade_type" == "UPGRADE_APP" ]
```

Comment:
Isn't this the same condition as before which already contains the same patch about the imgur thing ?

Reply:
Indeed, the code was duplicated

Reply:
Good catch!